pieces/simple_csv/rust/src/main.rs

```diff
@@ -0,0 +1,121 @@
+use serde::{Deserialize, Serialize};
+use std::fs;
+use std::path::Path;
+
+#[derive(Debug, Deserialize)]
+#[allow(dead_code)]
+struct CityTemps {
+    city: String,
+    country: String,
+    avg_temp_jan_f: f32,
+}
+
+#[derive(Debug, Serialize)]
+#[allow(dead_code)]
+struct CityTempsModified {
+    city: String,
+    country: String,
+    avg_temp_jan_f: f32,
+    avg_temp_jan_c: f32,
+}
+
+// impl block to convert temp from F to C
+impl CityTempsModified {
+    fn get_celsius(city_temps: &CityTemps) -> Self {
+        let city = &city_temps.city;
+        let country = &city_temps.country;
+        let avg_temp_jan_f = city_temps.avg_temp_jan_f;
+        CityTempsModified {
+            city: city.to_string(),
+            country: country.to_string(),
+            avg_temp_jan_f,
+            avg_temp_jan_c: convert_f_to_c(avg_temp_jan_f),
+        }
+    }
+}
+
+fn convert_f_to_c(temp_f: f32) -> f32 {
+    // Convert from Fahrenheit to Celsius to 1 decimal precision
+    let temp_c = (temp_f - 32.0) * 5.0 / 9.0;
+    (temp_c * 10.0).round() / 10.0
```

Comment:
Is there a better way to round to 1 decimal precision than this in Rust?

Reply:
Yup, couldn't find anything else better despite my extensive searches. Let's keep this for now.